src/js/editor/editor.js

```diff
@@ -339,6 +340,12 @@ class Editor {
           return;
         }
       }
+
+      // Above logic might delete redundant range, so callback must run after it.
+      let options = { insertNewline: true };
```

Comment:
@eguitarz Instead of using this ad hoc options hash, let's use a synthetic `event` object that has a `preventDefault` method on it. In user code, the developer can call the `preventDefault` method on it, e.g.:
```javascript
editor.willHandleNewline((event) => {
  if (shouldPreventDefault) {
    event.preventDefault();
  }
});
```